.github/workflows/main.yml

```diff
@@ -59,3 +59,41 @@ jobs:
       - name: Run pytest
         run: |
           pytest -v test/
+
+  lint:
+    runs-on: ubuntu-22.04
+    name: Lint
+
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # This is necessary to get the commits
+
+      - name: Get changed python files between base and head
+        run: >
+          echo "CHANGED_FILES=$(echo $(git diff --name-only ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} -- | grep \.py))" >> $GITHUB_ENV
+
+      - if: env.CHANGED_FILES
+        name: Set up Python
+        uses: actions/setup-python@master
+        with:
+          python-version: "3.10"
+
+      - if: env.CHANGED_FILES
+        name: Install Python packages
+        run: |
+          pip install -r requirements-dev.txt
```

Comment:
I think it should be `requirements.txt` rather than the `-dev` one.  The linter tools etc. are installed later.

Reply:
Files under `test` are using packages from `requirements-dev.txt`. Hence need to install them as well.

Reply:
Ah OK, that's a difference with `kernelci-core` where the main `requirements.txt` file has everything to run unit tests as well e.g. `pytest`.  I'm not sure what's the general convention for things that should go in `-dev.txt`.

Reply:
So `requirements.txt` has `pylint==2.12.2` here.  Which version of pylint is used by the annotation action?

Reply:
Interestingly, the annotation action installs `pylint==2.12.1`.

Reply:
Not sure how to debug that.  For now it shouldn't matter as they're very close versions but if for any reason we start running different pylint versions or other linter tools then we could start having different errors being reported which can be inconvenient.

Maybe `requirements-dev.txt` could be split into one file with the unit tests requirements and another one with the linters?  Say `requirements-tests.txt` which would include `requirements.txt` and add things like `pytest` and `fakeredis`, then `requirements-dev.txt` which would include `requirements-tests.txt` and add `pylint` and `pycodestyle` on top.  Then the annotations would be using `requirements-tests.txt`.

Reply:
The general convention is requirements for testing and development go to `requirements-dev.txt`. Still, there are no hard rules for that.

Reply:
Yes, we need to resolve the issues with version conflicts. 
It is a good idea to split requirements into multiple files.
How about adding `pylint` and `pycodestyle` to `requirements-linter.txt` file? Then the linter requirements file will include `requirements-tests.txt`.